richtext-ui/src/commonMain/kotlin/com/halilibo/richtext/ui/string/Text.kt

```diff
@@ -50,10 +50,10 @@ public fun RichTextScope.Text(
       overflow = overflow,
       maxLines = maxLines,
       isOffsetClickable = { offset ->
-        annotated.getConsumableAnnotations(text.formatObjects, offset).any()
+        annotated.getConsumableAnnotations(text.formatObjects, offset.coerceIn(0, annotated.length - 1)).any()
```

Comment:
offset was 1 when clicking the right half of the image. By coercing offset into the string length, the entire child content of a link becomes clickable. @halilozercan is going to do a sanity check but I couldn't find any new issues that this caused.

Reply:
This looks ok. The problem stems from `getOffsetForPosition` call which can return the length of the text to signify where the caret should be placed. Coercing the offset into actual character indices range should be ok in Image scenario. The only issue I can see with this approach is links that are placed at the very end of a paragraph. In that case, clicking end padding (the place where the text ends but text layout rectangle still occupies the screen) can trigger a link click. That may not be ideal.